pandas/io/formats/format.py

```diff
@@ -702,7 +733,7 @@ def _to_str_columns(self) -> List[List[str]]:
         """
         # this method is not used by to_html where self.col_space
         # could be a string so safe to cast
-        self.col_space = cast(int, self.col_space)
+        col_space = {k: cast(int, v) for k, v in self.col_space.items()}
```

Comment:
Is the cast still required here?

Reply:
@WillAyd  Not an actual cast, typing.cast just used to signal to the type checker (mypy). In this case self.col_space can be of type dict<str> by definition but we are sure that it is dict<int> within this scope